### PR TITLE
MAT-6106: Library Name 64 Character Limit

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-6106](https://jira.cms.gov/browse/MAT-6106)
(Optional) Related Tickets:
https://github.com/MeasureAuthoringTool/madie-layout/pull/263
https://github.com/MeasureAuthoringTool/madie-measure/pull/433

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editCqlLibrary/EditCqlLibrary.tsx [hunk 1]
```diff
@@ -408,7 +408,11 @@ const EditCqlLibrary = () => {
                       Boolean(formik.errors.cqlLibraryName)
                     }
                     {...formik.getFieldProps("cqlLibraryName")}
-                    helperText={formikErrorHandler("cqlLibraryName", true)}
+                    helperText={
+                      formikErrorHandler("cqlLibraryName", true) ||
+                      (formik.values.cqlLibraryName?.length > 64 &&
+                        "Libary Name cannot be more than 64 characters")
```
Comment (adongare) [inline src/components/editCqlLibrary/EditCqlLibrary.tsx]: there is a typo in `Libary`. also missing a period at the end of the sentence.

File: src/validators/CqlLibrarySchemaValidator.ts [hunk 2]
```diff
@@ -8,7 +8,8 @@ export const CqlLibrarySchemaValidator = Yup.object().shape({
     .matches(
       /^[A-Z][a-zA-Z0-9]*$/,
       "Library name must start with an upper case letter, followed by alpha-numeric character(s) and must not contain spaces or other special characters."
-    ),
+    )
+    .max(64, "Library name cannot be more than 64 characters long."),
```
Comment (adongare) [inline src/validators/CqlLibrarySchemaValidator.ts]: I would suggest just copying and pasting the message from the ticket.

File: src/components/common/CreateNewLibraryDialog.tsx [hunk 3]
```diff
@@ -195,12 +195,12 @@ const CreateNewLibraryDialog: React.FC<TestProps> = ({
               data-testid="cql-library-name-text-field"
               inputProps={{
                 "data-testid": "cql-library-name-text-field-input",
+                maxlength: 64,
               }}
               helperText={
                 (formik.touched["cqlLibraryName"] ||
                   focusedField === "cqlLibraryName") &&
-                (formikErrorHandler("cqlLibraryName") ||
-                  "Library name must start with an upper case letter, followed by alpha-numeric character(s) and must not contain spaces or other special characters.")
```
Comment (adongare) [inline src/components/common/CreateNewLibraryDialog.tsx]: @mcmcphillips is this related to 508? could you confirm removing this would not cause any issue?

Comment (ethankaplan) [reply src/components/common/CreateNewLibraryDialog.tsx]: I misinterpreted this code's purpose, added it back

**Review by jkotanchik-SB — DISMISSED** [review]

**Review by adongare — APPROVED** [review]
